parser-typechecker/unison/Main.hs

```diff
@@ -122,40 +137,62 @@ main = do
           theCodebase <- FileCodebase.getCodebaseOrExit mcodepath
           let fileEvent = Input.UnisonFileChanged (Text.pack "<standard input>") contents
           launch currentDir configFilePath theCodebase [Left fileEvent, Right $ Input.ExecuteI mainName, Right Input.QuitI]
-    "transcript" : args -> runTranscripts False mcodepath args
-    "transcript.fork" : args -> runTranscripts True mcodepath args
+    "transcript" : args' -> 
+      case args' of
+      "-preserve" : transcripts -> runTranscripts False True mcodepath transcripts
+      _                         -> runTranscripts False False mcodepath args'
+    "transcript.fork" : args' -> 
+      case args' of
+      "-preserve" : transcripts -> runTranscripts True True mcodepath transcripts
+      _                         -> runTranscripts True False mcodepath args'
```

Comment:
Hehe already feeling a little Boolean-blind here, though I'm not feeling creative in coming up with a short, clear type name we could use instead of Bool; just an idea, not a requirement.